### PR TITLE
Added torch implementation

### PR DESCRIPTION
**Pull request by martinezdavid12**
Added PyTorch implementation of core source files:

**Components:**
- **Models**: InterlacerResidualModel, ConvResidualModel, ConvNoResidualModel
- **Layers**: Interleaved, BatchNormConv, PiecewiseReLU, Mix
- **Utilities**: Complex-valued data handling and FFT operations

### CODE REVIEW

File: torch_interlacer/torch_interlacer/models.py [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+
+        out = self.out_conv(out)
+        out = out + x
+        if self.enforce_dc:
+            assert dc_mask is not None
+            out = dc_mask * x + (1 - dc_mask) * out
+        return out
+      
+# getters
+def get_conv_no_residual_model(
```
Comment (nalinimsingh) [inline torch_interlacer/torch_interlacer/models.py]: Redundant -- you can just get rid of this and call the Model classes directly

File: torch_interlacer/torch_interlacer/models.py [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+def get_conv_residual_model(input_size, 
+  nonlinearity, 
+  kernel_size, 
+  num_features, 
+  num_layers, 
+  enforce_dc=False
+  ):
+  return ConvResidualModel(input_size, nonlinearity, kernel_size, num_features, num_layers, enforce_dc=False)
+    
+def crop_320(inputs):
```
Comment (nalinimsingh) [inline torch_interlacer/torch_interlacer/models.py]: Untested/unnecessary?

**Review by nalinimsingh — CHANGES_REQUESTED** [review]
Thanks! As discussed in our meeting:
- Remove all untested functions
- Remove all debugging traces and unnecessary comments
- Try to consolidate the requirements.txt as much as possible
- Add data generator as its own module
- Remove the unnecessary getters and just use the model classes
- Make the README.md to point to the ipynb as an example of the workflow
- Update the outer-level README of higher-level repo to mention that there is also a torch implementation
